enhancements/insights/pulling-data-from-ocm.md

```diff
@@ -0,0 +1,139 @@
+- Periodically pull the data from the OCM API and expose it in the OpenShift API
+- This is an opt-in feature by a cluster user and might be moved to a different OCP component in the future
+
+### Non-Goals
+
+- Insights Operator providing any transformation or post-processing of the SCA certs pulled
+  from the OCM API
+
+## Proposal
+
+### User Stories
+
+#### Consume SCA certs exposed in the API
+
+As an OpenShift user
+I want to consume SCA certs to be able to consume RHEL content and to build
+corresponding container images.
+
+### Risks and Mitigations
+
+#### OCM API is down
+
+Risk: OCM API is down or doesn't provide up to date data.
+
+Risk: Insights Operator is unable to expose/update the data in the OpenShift API
+
+Mitigation: Introduce a new state in the Insights Operator (e.g "SCADataDegraded") and
+create a new alert based on this new state.
+
+## Design Details
+
+### Authorization
+
+The Insights Operator is able to pull the data from the OCM API using the existing `cloud.openshift.com` token
+available in the `pull-secret` (in the `openshift-config-managed` namespace).
+
+The Insights Operator must provide a cluster ID as an identifier of the cluster.
+
+### Data in API
+
+The SCA certificate is available via the `etc-pki-entitlement` secret in the `openshift-config-managed` namespace.
+
+### Update period
+- Insights Operator query the OCM API every 8 hours and downloads the full data provided
+
+### Test Plan
+
+- `insights-operator-e2e-tests` suite can verify the SCA cert data
+  is available
+- Basic test of the validity of the SCA certs. Mount the `etc-pki-entitlement` secret and run e.g `yum install` in the container
+
+### Graduation Criteria
+
+This feature is planned as a technical preview in OCP 4.9 and is planned to go GA in 4.10.
+
+#### Dev Preview -> Tech Preview
+- opt-in feature (called `InsightsOperatorPullingSCA`) enabled with `TechPreviewNoUpgrade` feature set
+- Insights Operator is able to download the data from OCM API and expose it in a cluster API
+- basic functionality is tested
+- this new functionality is documented
+
+#### Tech Preview -> GA
+- ability to distinguish various error states - e.g organization doesn't have SCA allowed versus API returns an error
+- inform a cluster user about the error state (problem with pulling the certificates)
+- the feature might be moved to a different OCP component
+
+#### Removing a deprecated feature
+
+The periodical data pulling can be easily disabled in the cluster configuration. Removing this feature will require updating the Insights operator code base and will remove the `etc-pki-entitlement` secret from the `openshift-config-managed` namespace.
+
+### Upgrade / Downgrade Strategy
+
+There is no upgrade/downgrade strategy needed.
+
+### Version Skew Strategy
+
+There is no Skew strategy needed. This work should have no impact on the upgrade. It doesn't require any coordinated behavior in the control plane. No other components will change.
```

Comment:
Does the API used to fetch the data ever change? Does the format of the cert ever change?

Reply:
Can you help me @petli-openshift to answer this question pls? 
I think the format of the cert is irrelevant from the perspective of this EP (iow it will expose what it receives). Pls correct me if I am wrong.

Reply:
The API to fetch the data does not change.

Reply:
The cert returned by RHSM is composed of several PEM-encoded blocks of data. The certificate format (what extensions are used, etc) is versioned, but this has not changed in many years. A certificate version change would only impact client tooling like subscription-manager and would be transparent to OCM.

Reply:
Thank you. I updated the section at https://github.com/openshift/enhancements/pull/683/files#diff-93dffd2aa59c973a916306675f62907275cff29ff9a16ba77aaf98cfa830249fR140